### PR TITLE
IAM role w/ policy

### PR DESCRIPTION
**Pull request by Tina-repo**

### CODE REVIEW

File: aws-gitlab-main.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#ROLE W/ IMBEDDED POLICY
+  GitLabS3Access: #Name of the role 
+    Type: 'AWS::IAM::Role'
+    Properties:
+      AssumeRolePolicyDocument: 
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com #?
```
Comment (sidewayset) [inline aws-gitlab-main.yml]: remove comment

File: aws-gitlab-main.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                - Effect: Allow
+                  Action:
+                  - 's3:AbortMultipleUpload'
+                  - 's3:CompletemultipartUpload'
+                  - 's3:ListBucket'
+                  - 's3:PutObject'
+                  - 's3:GetObject'
+                  - 's3:DeleteObject'
+                  - 's3:PutObjectAcl' 
+                  Resource: 'arn:aws:s3:::aws-gitlab-bucket'
```
Comment (sidewayset) [inline aws-gitlab-main.yml]: where is this bucket created? can we include it as part of the template and do !Ref , if not parameterized the name

File: aws-gitlab-main.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  RootInstanceProfile:
```
Comment (sidewayset) [inline aws-gitlab-main.yml]: name the logical resource same as role one but add InstnaceProfile to the name so its confusing

**Review by sidewayset — CHANGES_REQUESTED** [review]

File: aws-gitlab-main.yml (hunk 2 above)
Comment (Tina-repo) [reply aws-gitlab-main.yml]: I created it in my AWS. Believe I can change it to !Ref. Would I create the bucket with no specified properties? If you wanted to give a BucketName how would you know if the name is taken or not? What do you mean by parameterized the name?

Comment (sidewayset) [reply aws-gitlab-main.yml]: If bucket can be created inside the yaml template do that and do not do it in console. Once the final template is pushed the whole thing should go live like magic :) 

Bucket name: 
1. Either make the name dynamic with  unique element  such as !Sub gitlab-bucket-${AWS::AccountId} << unique element means nobody using it. Or do not give bucket a name in the yaml and aws will generate random string for it using LogicalResourceName of the bucket + random string

Bucket will have properties that is encrypting it properly. Potentially versioning and logging enabled depends. you can start with simple bucket w/ default encryption than go more on it

**Review by sidewayset — CHANGES_REQUESTED** [review]

File: aws-gitlab-main.yml [hunk 4]
```diff
@@ -24,7 +24,9 @@ Resources:
                   - 's3:GetObject'
                   - 's3:DeleteObject'
                   - 's3:PutObjectAcl' 
-                  Resource: 'arn:aws:s3:::aws-gitlab-bucket'
+                  Resource: 
+                    gl-s3-bucket:
```
Comment (sidewayset) [inline aws-gitlab-main.yml:28]: bruh tf? this was new one i have not seen before.

Great the bucket as another resource like the role, the profile and !Ref on the bucket for Resource piece of the role

**Review by sidewayset — CHANGES_REQUESTED** [review]

**Pull request closed without merging**